src/rmind/components/objectives/policy.py

```diff
@@ -152,37 +151,48 @@ def predict(
             )
 
             logits = self.heads.forward(features, batch_size=[b, 1])
+            if set(logits.keys()) != {Modality.DISCRETE, Modality.CONTINUOUS}:  # pyright: ignore[reportArgumentType]
+                msg = f"Only {Modality.DISCRETE} and {Modality.CONTINUOUS} modalities are supported, got {logits.keys()}"
+                raise ValueError(msg)
```

Comment:
perhaps define inline funcs with match/case that are then passed to `named_apply` vs enforcing here?